pango/src/functions.rs

```diff
@@ -27,13 +28,15 @@ pub fn shape_full(
         Some(s) => s.len(),
         None => 0,
     } as i32;
-    let paragraph_text = paragraph_text.to_glib_none();
     let item_length = item_text.len() as i32;
     unsafe {
+        // The function does not take null-terminated strings when a length is provided.
+        // It also requires item_text to point to a subsequence of paragraph_text.
+        // Using to_glib_none() on &str will copy the string and cause problems.
         ffi::pango_shape_full(
-            item_text.to_glib_none().0,
+            item_text.as_bytes().to_glib_none().0 as *const c_char,
```

Comment:
```suggestion
            item_text.as_ptr() as *const c_char,
```

seems simpler. Can you update that? Otherwise seems all correct, thanks!